build-aux/com.obsproject.Studio.json

```
@@ -19,7 +19,11 @@
         "--talk-name=org.a11y.Bus",
         "--own-name=org.kde.StatusNotifierItem-2-2",
         "--system-talk-name=org.freedesktop.Avahi",
-        "--env=VST_PATH=/app/extensions/Plugins/vst"
+        "--env=CLAP_PATH=$HOME/.clap:/app/extensions/Plugins/clap",
+        "--env=LADSPA_PATH=$HOME/.ladspa:/app/extensions/Plugins/ladspa",
+        "--env=LV2_PATH=$HOME/.lv2:/app/extensions/Plugins/lv2:/app/lib/lv2",
+        "--env=VST_PATH=$HOME/.vst:/app/extensions/Plugins/vst",
+        "--env=VST3_PATH=$HOME/.vst3:/app/extensions/Plugins/vst3"
```

```suggestion
        "--env=CLAP_PATH=/app/extensions/Plugins/clap",
        "--env=LADSPA_PATH=/app/extensions/Plugins/ladspa",
        "--env=LV2_PATH=/app/extensions/Plugins/lv2:/app/lib/lv2",
        "--env=VST_PATH=/app/extensions/Plugins/vst",
        "--env=VST3_PATH=/app/extensions/Plugins/vst3"
```

Even if it looks like a good idea, it is not. `$HOME` must be be considered inaccessible and so the Flatpak rely only on extensions.

Users are free to modify the Flatpak permission after installation, but by default it must not have access.

---

it  is the spec that recommends using $HOME here, see the LV2 portion for https://github.com/flathub/org.freedesktop.LinuxAudio.BaseExtension

---

also IMO not having access to the rest of the plugins the user has installed is a pretty hard limitation.
users will install some plugins from a random vendor, which will instruct them to place them in the `~/.vst3` folder, and then be confused why OBS cannot find them.

it is suboptimal but we could make the case for plugins installed in /usr/lib/vst3 to not be available inside the flatpak container, but also doing the same for plugins the user has manually installed is a very bad user experience.

---

This is how sandboxing works, there is some isolation and there is no plan to punch more hole in the Flatpak sandbox.
Like we don't support manual installation of obs plugins for the Flatpak.

> it is suboptimal but we could make the case for plugins installed in /usr/lib/vst3 to not be available inside the flatpak container,

This is on purpose, those plugins were built for the host not the Flatpak runtime.

---

> > it is suboptimal but we could make the case for plugins installed in /usr/lib/vst3 to not be available inside the flatpak container,
> 
> This is on purpose, those plugins were built for the host not the Flatpak runtime.

that is exactly my point, we can make the case for why we do not want those plugins listed. but that case is invalid for plugins on the $HOME dir.

if someone wants to use any commercial plugin on OBS that is not packaged in flathub (which are most of them) without access to $HOME/.lv2 or alike, there is no way to get these plugins to work.

---

> that is exactly my point, we can make the case for why we do not want those plugins listed. but that case is invalid for plugins on the $HOME dir.

`$HOME` must stay out of those, I repeat that there is no plan to punch more hole in the Flatpak sandbox.

> if someone wants to use any commercial plugin on OBS that is not packaged in flathub (which are most of them) without access to $HOME/.lv2 or alike, there is no way to get these plugins to work.

There is, it's called modifying the Flatpak permissions with the `flatpak` command or program like Flatseal.
`$HOME` will not be allowed out of the box by OBS Studio Flatpak.

---

errr okay, but still think it is quite a silly thing to do here. if users need to hack their way into a package just to use plugins they bought, they will just most likely not use the flatpak version to begin with, too much hassle.

---

sorry but I went to research a bit on these properties, looking how other DAWs and plugin hosts were handling this, and then noticed the current OBS package contains this:

```
--filesystem=host
```

this grants full access to the host filesystem, tested locally and indeed OBS can load any local file be it from $HOME or /usr.
so trying to restrict the plugin paths to not include user-installed plugins under $HOME makes no sense to me.

my current code for this OBS carla plugin includes the option to load an audio plugin from disk (say someone wants to load a DLL in a custom location).
I am currently able to load plugin files under flatpak OBS  from /usr/lib/vst/ without a problem.
Listing the entire contents of $HOME is also doable without a problem either.

was that a lapse?
doesnt seem like it since this full system access has been there since the first push https://github.com/obsproject/obs-studio/commit/036012d32954b9e1b6c243431e8064988516c4e5

---

The scope of our needs for such a plugin is limited to VST3 right now. _Anything else_ added on top of that should be removed or omitted if it leads to complications (such as LV2 being the add one out requiring `$HOME` here).

---

`--filesystem=host` is here because sources, recordings still need to punched hole in the sandbox, in the long term those hole will be removed.

So no `$HOME` for audio plugins paths out of the box.